test/mz-e2e/mzcompose.py

```diff
@@ -0,0 +1,254 @@
+USERNAME = os.getenv("NIGHTLY_MZ_USERNAME", "infra+bot@materialize.com")
+APP_PASSWORD = os.environ["MZ_CLI_APP_PASSWORD"]
+VERSION = "devel-" + os.environ["BUILDKITE_COMMIT"]
+
+SERVICES = [
+    Cockroach(setup_materialize=True),
+    Materialized(
+        # We use materialize/environmentd and not materialize/materialized here
+        # in order to ensure a perfect match to the container that should be
+        # deployed to the cloud
+        image=f"materialize/environmentd:{VERSION}",
+        external_cockroach=True,
+        persist_blob_url="file:///mzdata/persist/blob",
+        options=[
+            "--orchestrator-process-secrets-directory=/mzdata/secrets",
+            "--orchestrator-process-scratch-directory=/scratch",
+        ],
+    ),
+    Testdrive(),  # Overriden below
+    Mz(
+        region=REGION,
+        environment=ENVIRONMENT,
+        app_password=APP_PASSWORD,
+    ),
+]
+
+
+def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
+    """Deploy the current source to the cloud and run tests."""
+
+    parser.add_argument(
+        "--cleanup",
+        default=True,
+        action=argparse.BooleanOptionalAction,
+        help="Destroy the region at the end of the workflow.",
+    )
+    parser.add_argument(
+        "--version-check",
+        default=True,
+        action=argparse.BooleanOptionalAction,
+        help="Perform a version check.",
+    )
+
+    parser.add_argument(
+        "td_files",
+        nargs="*",
+        default=["*.td"],
+        help="run against the specified files",
+    )
+
+    args = parser.parse_args()
+
+    if args.cleanup:
+        workflow_disable_region(c)
+
+    test_failed = True
+    try:
+        print(f"Enabling region using Mz version {VERSION} ...")
+        try:
+            c.run("mz", "region", "enable", "--version", VERSION)
+        except UIError:
+            # Work around https://github.com/MaterializeInc/materialize/issues/17219
+            pass
+
+        time.sleep(10)
+
+        assert "materialize.cloud" in cloud_hostname(c)
+        wait_for_cloud(c)
+
+        # if args.version_check:
+        #     version_check(c)
+
+        # Test - `mz app-password`
+        # Assert `mz app-password create`
+        new_app_password_name = "Materialize CLI (mz) - Nightlies"
+        output = c.run(
+            "mz", "app-password", "create", new_app_password_name, capture=True
+        )
+        assert "mzp_" in output.stdout
+
+        # Assert `mz app-password list`
+        output = c.run("mz", "app-password", "list", capture=True)
+        assert new_app_password_name in output.stdout
+
+        # // Test - `mz secrets`
+        secret = "decode('c2VjcmV0Cg==', 'base64')"
+        output = c.run(
+            "mz", "secret", "create", "CI_SECRET", stdin=secret, capture=True
+        )
+        assert output.returncode == 0
+
+        output = c.run("mz", "sql", "--", "-q", "-c", "SHOW SECRETS;", capture=True)
```

Comment:
We have had regressions in the past where a secret would appear in `SHOW SECRETS` but then could not actually be used to create a database object. So I suggest you create a secret that actually exists in the real world and use it to connect to a real source. In Buildkite, `CONFLUENT_API_KEY` and `CONFLUENT_API_SECRET` exist and you can use them to connect to an actual Kafka source -- see `test/cloud-canary/canary-sources.td` for an example.